tools/autograd/templates/VariableType.cpp

```diff
@@ -238,29 +239,36 @@ static void ensure_no_aten_scalars(Tensor & data) {
   }
 }
 
-template<typename T>
-static bool computes_grad_tmpl(T tensors) {
-  if (!GradMode::is_enabled()) {
-    return false;
-  }
+using TensorRef = std::reference_wrapper<const Tensor>;
+using TensorRefList = std::initializer_list<TensorRef>;
```

Comment:
Can you make that an `ArrayRef`? That's what `*List` usually is in our code

Reply:
Actually are they even used anywhere?

Reply:
No, they're not used anywhere.

Reply:
These were carefully defined so as to avoid unnecessary atomic increments and decrements.